src/Elasticsearch.Net/Configuration/ConnectionConfiguration.cs

```diff
@@ -502,6 +507,12 @@ public T EnableDebugMode(Action<IApiCallDetails> onRequestCompleted = null)
 		public T SkipDeserializationForStatusCodes(params int[] statusCodes) =>
 			Assign(new ReadOnlyCollection<int>(statusCodes), (a, v) => a._skipDeserializationForStatusCodes = v);
 
+		/// <summary>
+		/// The user agent string to send with requests. Useful for debugging purposes to understand client and framework
+		/// versions that initiate requests to Elasticsearch
+		/// </summary>
+		public T UserAgent(string userAgent) => Assign(userAgent, (a, v) => a._userAgent = v);

```

Comment:
Should we allow the full userAgent to be configured or just the moniker?

Bonus would be that `ConnectionSettings` and `ConnectionConfiguration` could ship different default `monikers` but rely on the same full agent string building.

Reply:
The client identification is part of the metadata. Perhaps this public method should only set the application name (where we put repos name).